lib/ddtrace/contrib/configuration/settings.rb

```diff
@@ -29,6 +32,21 @@ def [](name)
         def []=(name, value)
           respond_to?("#{name}=") ? send("#{name}=", value) : set_option(name, value)
         end
+
+        DEPRECATION_WARNING = %(
+          Explicitly providing a tracer instance is DEPRECATED.
+          It's recommended to not provide an explicit tracer instance
+          and let Datadog::Contrib::Configuration::Settings resolve
+          the correct tracer internally.
+          ).freeze
+
+        include Datadog::Patcher # DEV includes #do_once here. We should move that logic to a generic component.
```

Comment:
Totally agree... before we had started this refactor, I had been hoping to refactor how we do deprecation warnings. e.g. Extract all the necessary functions to track and raise deprecation warnings into a module.

This is fine for the short term but we will want to get rid of this.